docs/architecture/adr-043-interchain-identifiers.md

```diff
@@ -0,0 +1,108 @@
+# ADR 043: Interchain Identifiers
```

Comment:
```suggestion
# ADR 045: Interchain Identifiers
```

Reply:
Looks like this will bump to 046 if we get this merged soon.